source/CAS/Client.php

```diff
@@ -2387,23 +2428,39 @@ public function setCallbackURL($url)
     private function _callback()
     {
         phpCAS::traceBegin();
-        if (preg_match('/^PGTIOU-[\.\-\w]+$/', $_GET['pgtIou'])) {
-            if (preg_match('/^[PT]GT-[\.\-\w]+$/', $_GET['pgtId'])) {
-                $this->printHTMLHeader('phpCAS callback');
-                $pgt_iou = $_GET['pgtIou'];
-                $pgt = $_GET['pgtId'];
-                phpCAS::trace('Storing PGT `'.$pgt.'\' (id=`'.$pgt_iou.'\')');
-                echo '<p>Storing PGT `'.$pgt.'\' (id=`'.$pgt_iou.'\').</p>';
-                $this->_storePGT($pgt, $pgt_iou);
-                $this->printHTMLFooter();
+        if ($this->_isCallbackModeUsingPost()) {
+            $pgtId = $_POST['pgtId'];
+            $pgtIou = $_POST['pgtIou'];
+        } else {
+            $pgtId = $_GET['pgtId'];
+            $pgtIou = $_GET['pgtIou'];
+        }
+        if (preg_match('/^PGTIOU-[\.\-\w]+$/', $pgtIou)) {
+            if (preg_match('/^[PT]GT-[\.\-\w]+$/', $pgtId)) {
+                phpCAS::trace('Storing PGT `'.$pgtId.'\' (id=`'.$pgtIou.'\')');
+                $this->_storePGT($pgtId, $pgtIou);
+                if (array_key_exists('HTTP_ACCEPT', $_SERVER) &&
+                    (   $_SERVER['HTTP_ACCEPT'] == 'application/xml' ||
+                        $_SERVER['HTTP_ACCEPT'] == 'text/xml'
+                    )
+                ) {
```

Comment:
And this is where `CallbackFormat` should go I guess? (If your intention is to make this output format adjustable)

Reply:
It would go on line 2431, replacing 
`if ($this->_isCallbackModeUsingPost()) {`
with the new naming.

Indeed the intention is to make the output format adjustable; but not by the developer; by the library itself. If it receives the callback request using GET (the normal) case, it does process it as usual. If it receives it as POST (the new use case); then it sets a flag (`$_callback_mode_using_post`); this flag is later used in `function _callback()` (just for the sake of avoiding detection if GET or POST in several places)

Reply:
Thank you for the explanation! Then I believe this doesn't need immediate change anymore :-)